pkg/asset/agent/agentconfig/agent_config.go

```diff
@@ -0,0 +1,112 @@
+package agentconfig
+
+import (
+	"fmt"
+	"os"
+
+	"github.com/openshift/installer/pkg/asset"
+	"github.com/openshift/installer/pkg/types/agent"
+	"github.com/pkg/errors"
+	"k8s.io/apimachinery/pkg/util/validation/field"
+	"sigs.k8s.io/yaml"
+)
+
+var (
+	agentConfigFilename = "agent-config.yaml"
+)
+
+// Asset reads the agent-config.yaml file.
+type Asset struct {
+	File   *asset.File
+	Config *agent.Config
+}
+
+// Name returns a human friendly name for the asset.
+func (*Asset) Name() string {
+	return "Agent Config"
+}
+
+// Dependencies returns all of the dependencies directly needed to generate
+// the asset.
+func (*Asset) Dependencies() []asset.Asset {
+	return []asset.Asset{}
+}
+
+// Generate generates the Agent manifest.
+func (a *Asset) Generate(dependencies asset.Parents) error {
+	return nil
+}
+
+// Files returns the files generated by the asset.
+func (a *Asset) Files() []*asset.File {
+	if a.File != nil {
+		return []*asset.File{a.File}
+	}
+	return []*asset.File{}
+}
+
+// Load returns agent config asset from the disk.
+func (a *Asset) Load(f asset.FileFetcher) (bool, error) {
+
+	file, err := f.FetchByName(agentConfigFilename)
+	if err != nil {
+		if os.IsNotExist(err) {
+			return false, nil
+		}
+		return false, errors.Wrap(err, fmt.Sprintf("failed to load %s file", agentConfigFilename))
+	}
+
+	config := &agent.Config{}
+	if err := yaml.UnmarshalStrict(file.Data, config); err != nil {
+		return false, errors.Wrapf(err, "failed to unmarshal %s", agentConfigFilename)
+	}
+
+	a.File, a.Config = file, config
+	if err = a.finish(); err != nil {
+		return false, err
+	}
+
+	return true, nil
+}
+
+func (a *Asset) finish() error {
+	if err := a.validateAgent().ToAggregate(); err != nil {
+		return errors.Wrapf(err, "invalid Agent Config configuration")
+	}
+
+	return nil
+}
+
+func (a *Asset) validateAgent() field.ErrorList {
+	allErrs := field.ErrorList{}
+
+	if err := a.validateNodesHaveAtLeastOneMacAddressDefined(); err != nil {
+		allErrs = append(allErrs, err...)
+	}
+
+	return allErrs
+}
+
+func (a *Asset) validateNodesHaveAtLeastOneMacAddressDefined() field.ErrorList {
```

Comment:
nit: the name of the rule looks a little bit misleading: the validation checks that _all_ the hosts have an interface with a macaddress configured